packages/headless/src/features/configuration/configuration-slice.test.ts

```diff
@@ -199,6 +204,45 @@ describe('configuration slice', () => {
       ).toEqual(expectedState);
     });
 
+    it('clearCoveoAnalyticsClient when needed', () => {
+      const existingState = getConfigurationInitialState();
+      existingState.analytics.enabled = true;
+      configurationReducer(
+        existingState,
+        updateAnalyticsConfiguration({
+          enabled: false,
+        })
+      );
+
+      expect(clearCoveoAnalyticsClient).toHaveBeenCalledTimes(1);
+    });
+
+    it('does not clearCoveoAnalyticsClient when it is already disabled', () => {
+      const existingState = getConfigurationInitialState();
+      existingState.analytics.enabled = false;
+      configurationReducer(
+        existingState,
+        updateAnalyticsConfiguration({
+          enabled: false,
+        })
+      );
+
+      expect(clearCoveoAnalyticsClient).not.toHaveBeenCalled();
+    });
+
+    it('does not clearCoveoAnalyticsClient when it is enabled', () => {
+      const existingState = getConfigurationInitialState();
+      existingState.analytics.enabled = false;
+      configurationReducer(
+        existingState,
+        updateAnalyticsConfiguration({
+          enabled: true,
+        })
+      );
```

Comment:
For prosperity, shouldn't we have a test that checks that it does not clear when it is already enabled.